packages/ethereum-contracts/contracts/agreements/ConstantFlowAgreementV1.sol

```diff
@@ -105,39 +105,60 @@ contract ConstantFlowAgreementV1 is
      * IConstantFlowAgreementV1 interface
      *************************************************************************/
 
-    /// @dev IConstantFlowAgreementV1.getMaximumFlowRateFromDeposit implementation
-    function getMaximumFlowRateFromDeposit(
-        ISuperfluidToken token,
+    function getMaximumFlowRateFromDepositPure(
+        uint256 liquidationPeriod,
         uint256 deposit)
-        external view override
+        public pure
         returns (int96 flowRate)
     {
         require(deposit < 2**95, "CFA: deposit number too big");
         deposit = _clipDepositNumberRoundingDown(deposit);
-        (uint256 liquidationPeriod, ) = _decode3PsData(token);
+
         uint256 flowrate1 = deposit / liquidationPeriod;
 
         // NOTE downcasting is safe as we constrain deposit to less than
         // 2 ** 95 so the resulting value flowRate1 will fit into int96
         return int96(int256(flowrate1));
     }
 
+    function getDepositRequiredForFlowRatePure(
+        uint256 minimumDeposit,
+        uint256 liquidationPeriod,
+        int96 flowRate)
+        public pure
+        returns (uint256 deposit)
+    {
+        require(flowRate >= 0, "CFA: not for negative flow rate");
```

Comment:
observation: we require the flowrate to be positive here, although it's in general negative from a sender's perspective.  
May be less confusing if in such places (were a positive one is assumed despite sender perspective) we don't use signed ints.

Reply:
then there is a casting issue between uint96 to int96 flowRate, it is rather easier to keep the same data type, and do a runtime parameter validation.